lib/validators/specTransformer.ts

```diff
@@ -27,6 +28,28 @@ export function transform(spec: SwaggerObject): SwaggerObject {
   return spec
 }
 
+export interface TitleObject {
+  position?: FilePosition
+  url?: string
+  readonly title: string
+}
+
+const getFileInfo = (info: Info): FileInfo =>
+  info.kind === "file" ? info : getFileInfo(info.parent)
+
 function insertSchemaTitle(model: SchemaObject, title: string) {
-  model.title = title
+  const info = getInfo(model)
+  const titleObject: TitleObject = {
+    title
+  }
+  if (info !== undefined) {
+    if (info.kind === "object") {
+      titleObject.position = {
+        line: info.position.line + 1,
+        column: info.position.column + 1,
+      }
+    }
+    titleObject.url = getFileInfo(info).url
+  }
+  model.title = JSON.stringify(titleObject)
```

Comment:
why stringify ? and not just leave it as an object ?

Reply:
I've tried passing an object. In this case, schema validation starts failing.

Reply:
uhhh, i wonder why is that...ok